cli/azd/pkg/tools/azcli/ad.go

```diff
@@ -367,3 +372,68 @@ func (cli *azCli) createRoleAssignmentsClient(
 
 	return client, nil
 }
+
+// Get role definition id from role assignment within specific scope and principalID
+func (cli *azCli) getUserRoleDefinitionID(
```

Comment:
Since this method returns a list of role ids I would consider changing the function name to something like 'getUserRoleDefinitionIdList` or similar.